src/main/java/com/coreos/jetcd/lease/KeepAlive.java

```diff
@@ -1,65 +1,72 @@
 package com.coreos.jetcd.lease;
 
-import com.coreos.jetcd.Lease.LeaseHandler;
-import com.coreos.jetcd.api.LeaseGrantResponse;
+import com.coreos.jetcd.LeaseImpl.KeepAliveListenerImpl;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * The KeepAlive hold the keepAlive information for lease.
  */
 public class KeepAlive {
 
-  private final long leaseID;
-
-  private long ttl;
-
-  private LeaseGrantResponse leaseGrantResponse;
-
   private long deadLine;
 
   private long nextKeepAlive;
 
-  private LeaseHandler leaseHandler;
+  private List<KeepAliveListenerImpl> keepAliveListeners = new ArrayList<>();
 
-  public KeepAlive(long leaseID) {
-    this(leaseID, null);
-  }
+  private DoneHandler doneHandler;
 
-  public KeepAlive(long leaseID, LeaseHandler leaseHandler) {
-    this.leaseID = leaseID;
-    this.leaseHandler = leaseHandler;
+  @FunctionalInterface
+  public interface DoneHandler {
```

Comment:
It's not clear to me how this is useful; if the keepalive is closed by the user then the user already knows it's closing so there's no need for callback (or is close async? it probably shouldn't be...). If the connections close without user intervention, it's not obvious whether there were any error conditions.